pkg/testutils/e2e/kubernetes/namespace/namespace_test.go

```diff
@@ -0,0 +1,66 @@
+package namespace
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_CreateNamespace(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		title       string
+		options     []TestNamespaceOption
+		expectError bool
+		check       func(kubernetes.Interface, string) error
+	}{
+		{
+			title:       "default options",
+			options:     []TestNamespaceOption{},
+			expectError: false,
+			check: func(k8s kubernetes.Interface, ns string) error {
+				return nil
+			},
+		},
+		{
+			title:       "random namespace",
+			options:     []TestNamespaceOption{WithPrefix("prefix-")},
+			expectError: false,
+			check: func(k8s kubernetes.Interface, ns string) error {
+				return nil
```

Comment:
Should we check here that `strings.HasPrefix(ns, "prefix-")`?

Reply:
Good catch, I forgot to implement the checks :sweat_smile: